src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -871,133 +870,21 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
-
-    def load_from_store(self) -> None:
-        cfg = self.mgr._get_exporter_config()
-
-        assert isinstance(cfg, dict)
-        self.crt = cfg.get('crt', "")
-        self.key = cfg.get('key', "")
-        self.token = cfg.get('token', "")
-        self.port = cfg.get('port', "")
-
-    def load_from_json(self, json_str: str) -> Tuple[int, str]:
-        try:
-            cfg = json.loads(json_str)
-        except ValueError:
-            return 1, "Invalid JSON provided - unable to load"
-
-        if not all([k in cfg for k in CephadmExporterConfig.required_keys]):
-            return 1, "JSON file must contain crt, key, token and port"
-
-        self.crt = cfg.get('crt')
-        self.key = cfg.get('key')
-        self.token = cfg.get('token')
-        self.port = cfg.get('port')
-
-        return 0, ""
-
-    def validate_config(self) -> Tuple[int, str]:
-        if not self.ready:
-            return 1, "Incomplete configuration. cephadm-exporter needs crt, key, token and port to be set"
-
-        for check in [self._validate_tls, self._validate_token, self._validate_port]:
-            rc, reason = check()
-            if rc:
-                return 1, reason
-
-        return 0, ""
-
-    def _validate_tls(self) -> Tuple[int, str]:
-
-        try:
-            verify_tls(self.crt, self.key)
-        except ServerConfigException as e:
-            return 1, str(e)
-
-        return 0, ""
-
-    def _validate_token(self) -> Tuple[int, str]:
-        if not isinstance(self.token, str):
-            return 1, "token must be a string"
-        if len(self.token) < 8:
-            return 1, "Token must be a string of at least 8 chars in length"
-
-        return 0, ""
-
-    def _validate_port(self) -> Tuple[int, str]:
-        try:
-            p = int(str(self.port))
-            if p <= 1024:
-                raise ValueError
-        except ValueError:
-            return 1, "Port must be a integer (>1024)"
-
-        return 0, ""
-
-
-class CephadmExporter(CephadmService):
-    TYPE = 'cephadm-exporter'
+class CephfsMirrorService(CephService):
+    TYPE = 'cephfs-mirror'
 
     def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonDeploySpec:
         assert self.TYPE == daemon_spec.daemon_type
 
-        cfg = CephadmExporterConfig(self.mgr)
-        cfg.load_from_store()
-
-        if cfg.ready:
-            rc, reason = cfg.validate_config()
-            if rc:
-                raise OrchestratorError(reason)
-        else:
-            logger.info(
-                "Incomplete/Missing configuration, applying defaults")
-            self.mgr._set_exporter_defaults()
-            cfg.load_from_store()
-
-        if not daemon_spec.ports:
-            daemon_spec.ports = [int(cfg.port)]
+        ret, keyring, err = self.mgr.check_mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': self.get_auth_entity(daemon_spec.daemon_id),
+            'caps': ['mon', 'allow r',
```

Comment:
FYI, this may become `profile cephfs-mirror`. See also: https://github.com/ceph/ceph/pull/39657#pullrequestreview-597719978